### PR TITLE
239 axis scale with data

### LINKED ISSUE #239: Axis scaling with data
Currently our axes scale, but are statically initialized. We need to scale each axis based on the data that we have selected to plot.

### PR DESCRIPTION
**Pull request by LongQuanNguyen**
Closes #239

# What was the issue
- Make each axis scale the interval correctly based on the absolute max values, and round off to the next integer.
- Example: columnX's max value is -3.3. Then the recorded max value will be 4, the x-axis will display from -4 to 4.
# What was done and why
- Modify getPoints in the repository to return absolute-max-round-to-next-int values in an array.
- Modify createGraphingDataPoint and createPosition to take in this max values array.
- Modify each SingleAxis, GenerateXYZ, GenerateTicks to take in their correspond max values to display intervals appropriately.

# How it was tested
- Modify old test cases to match new implementation of functions.
- Add test file and more unit test case.
- Visually test with max values of 10, 100 and 1000 and the graph interval display correctly as below:
image
- Try plotting three points: (1, 10, 100), (5, 50, 500), (10, 100, 1000) and they seem to plot correctly:
image

### CODE REVIEW

File: my-webxr-app/src/App.tsx [hunk 1]
```diff
@@ -15,22 +12,19 @@ import GraphingDataPointMenu from './components/GraphingDataPointMenu';
 import { PointSelectionProvider } from './contexts/PointSelectionContext';
 import './styles.css';
 import TestingOptions from './smoketest/TestingOptions';
-import DataPoint from './repository/DataPoint';
 import { rollbarConfig } from './utils/LoggingUtils';
 
 // minNum and maxNum will be from the csv file, just hardcoded for now
-const minNum: number = -10;
-const maxNum: number = 10;
+const maxNum: Array<number> = [10, 100, 1000];
```
Comment (matthew-buglass) [inline my-webxr-app/src/App.tsx:18]: Do we want the graph to start with 10, 100, 1000? Or do we want them all to start the same?

File: my-webxr-app/src/components/CreatePointPositions.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import { Vector3 } from 'three';
+import assert from '../utils/Assert';
+
+/**
+ * Creates a Vector for the points position relative to the size of the Axis and the Data given to
+ * the point.
+ * @preconds - The maxData must be an array of three non-negative integers.
```
Comment (matthew-buglass) [inline my-webxr-app/src/components/CreatePointPositions.tsx:7]: @pre-condition
@post-condition

File: my-webxr-app/src/components/GenerateTicks.tsx [hunk 3]
```diff
@@ -3,84 +3,84 @@ import * as assert from 'assert';
...
  * @param {number} radius corner radius of the tick's shape
  * @param {number} label the number to display with the tick
- * @param {number} increment the number of increments away from the start of the axis
  * @param {string} axis the label of the axis we are displaying one ('x', 'y', or 'z')
+ * @param {number} maxValue the maximum value plotted point of the axis
  * @return {JSX.Element} the tick element and its number label
  * @constructor
  */
-export default function GenerateTick(
+export default function GenerateTicks(
```
Comment (matthew-buglass) [inline my-webxr-app/src/components/GenerateTicks.tsx:23]: This only creates a singular Tick. So we shouldn't have the name be plural.

File: my-webxr-app/src/components/SingleAxis.tsx [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import * as THREE from 'three';
-import GenerateTick from './CreateTick';
+import GenerateTick from './GenerateTicks';
```
Comment (matthew-buglass) [inline my-webxr-app/src/components/SingleAxis.tsx:2]: Tick again

File: my-webxr-app/src/components/SingleAxis.tsx [hunk 5]
```diff
@@ -11,13 +11,16 @@ interface SingleAxisProps {
...
   maxValue: number;
   axis: string;
 }
 
 /**
  * Creates a Single axis object
+ * Helper function to for GenerateXYZ()
+ *
+ * @pre-conditions - maxValue is a non-negative integer
+ *
```
Comment (matthew-buglass) [inline my-webxr-app/src/components/SingleAxis.tsx:23]: @post-condition

File: my-webxr-app/src/contexts/AxesSelectionContext.tsx [hunk 6]
```diff
@@ -0,0 +1,126 @@
+import React, {
+  createContext, useContext, useMemo, useState,
+} from 'react';
+
+// import * as log4js from "log4js";
```
Comment (matthew-buglass) [inline my-webxr-app/src/contexts/AxesSelectionContext.tsx:5]: This can be deleted

File: my-webxr-app/src/contexts/AxesSelectionContext.tsx [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+  const setSelectedXAxis = (
+    newValue: React.SetStateAction<string | null>,
+  ) => {
+    // log4js
+    //   .getLogger()
+    //   .debug(
+    //     "AxesSelectionContext: updating selectedXAxis state to " +
+    //       newValue,
+    //   );
```
Comment (matthew-buglass) [inline my-webxr-app/src/contexts/AxesSelectionContext.tsx:65]: Replace with Rollbar

File: my-webxr-app/src/contexts/AxesSelectionContext.tsx [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+
+  const setSelectedYAxis = (
+    newValue: React.SetStateAction<string | null>,
+  ) => {
+    // log4js
+    //   .getLogger()
+    //   .debug(
+    //     "AxesSelectionContext: updating selectedYAxis state to " +
+    //       newValue,
+    //   );
```
Comment (matthew-buglass) [inline my-webxr-app/src/contexts/AxesSelectionContext.tsx:77]: Replace with Rollbar

File: my-webxr-app/src/contexts/AxesSelectionContext.tsx [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+
+  const setSelectedZAxis = (
+    newValue: React.SetStateAction<string | null>,
+  ) => {
+    // log4js
+    //   .getLogger()
+    //   .debug(
+    //     "AxesSelectionContext: updating selectedZAxis state to " +
+    //       newValue,
+    //   );
```
Comment (matthew-buglass) [inline my-webxr-app/src/contexts/AxesSelectionContext.tsx:89]: Replace with Rollbar

File: my-webxr-app/tests/components/GenerateTicks.test.tsx [hunk 10]
```diff
@@ -1,6 +1,6 @@
 import ReactThreeTestRenderer, { waitFor } from '@react-three/test-renderer';
 import { Vector3 } from 'three';
-import GenerateTick from '../../src/components/CreateTick';
+import GenerateTick from '../../src/components/GenerateTicks';
```
Comment (matthew-buglass) [inline my-webxr-app/tests/components/GenerateTicks.test.tsx:3]: Tick

**Review by matthew-buglass — CHANGES_REQUESTED** [review]
There are a few changes that need to be made, but it looks good overall. I want to wait until Aesha's PR is merged since this is based off that one.

**Comment by matthew-buglass** [comment]
The alternative approach is to merge into Aesha's branch

File: my-webxr-app/src/App.tsx (hunk 1 above)
Comment (LongQuanNguyen) [reply my-webxr-app/src/App.tsx:18]: I should get rid of that. They were mean for visual confirmation, the max will be determined by the column data.

**Review by matthew-buglass — APPROVED** [review]

**Pull request merged**